syntax="proto3";
package pms;

option optimize_for = LITE_RUNTIME;

// the server cmd
enum EServerCmd{
    CLOGIN      = 0;
    CSNDMSG     = 1;
    CGETMSG     = 2;
    CLOGOUT     = 3;
    CKEEPALIVE  = 4;
    CSYNCSEQN   = 5;
    CSYNCDATA   = 6;
}

// different server module
// the module means different business
// it used between client and server
// it can tell server this client belong to which server
enum EModuleType{
    TINVALID0   = 0;
    TINVALID1   = 1;
    TMEETING    = 2;
    TP2P        = 3;
    TLIVE       = 4;
    TCALLCENTER = 5;
    TLOGICAL    = 6;
    TSEQUENCE   = 7;
    TSTORAGE    = 8;
}

// this is for request
message MsgReq{
    EServerCmd     svr_cmds = 1;
    EModuleType    mod_type = 2;
    bytes          content  = 3;
}

// this is for response
message MsgRep{
    EServerCmd     svr_cmds = 1;
    EModuleType    mod_type = 2;
    bytes          rsp_cont = 3;
    sint32         rsp_code = 4;
}

// users send to
message ToUser{
    repeated string users = 1;
}
